flutter/lib/models/model.dart

```diff
@@ -1789,6 +1832,59 @@ class ViewStyle {
   }
 }
 
+enum EdgeScrollState {
+  inactive,
+  armed,
+  active,
+}
+
+class EdgeScrollFallbackState {
+  final CanvasModel _owner;
+
+  late Ticker _ticker;
+
+  Duration _lastTotalElapsed = Duration.zero;
+  bool _nextEventIsFirst = true;
+  Vector2 _encroachment = Vector2.zero();
+
+  EdgeScrollFallbackState(this._owner, TickerProvider tickerProvider) {
+    _ticker = tickerProvider.createTicker(emitTick);
+  }
+
+  setEncroachment(Vector2 encroachment) {
+    _encroachment = encroachment;
+  }
+
+  emitTick(Duration totalElapsed) {
+    if (_nextEventIsFirst) {
+      _lastTotalElapsed = totalElapsed;
+      _nextEventIsFirst = false;
+    } else {
+      final thisTickElapsed = totalElapsed - _lastTotalElapsed;
+
+      const double FrameTime = 1000.0 / 60.0;
```

Comment:
Suggestion from the `Flutter` extension in vscode.

```
The constant name 'FrameTime' isn't a lowerCamelCase identifier.
Try changing the name to follow the lowerCamelCase style.
```


The following names may be better.

```dart
      const double kFrameTime = 1000.0 / 60.0;
      const double kSpeedFactor = 0.1;
```

Reply:
Ah, yes, I'd seen that elsewhere and it hadn't clicked that that's the convention for constants.